blockchain/block/block_deserializer.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2022 IoTeX Foundation
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package block
+
+import (
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+	"google.golang.org/protobuf/proto"
+)
+
+// Deserializer de-serializes a block
+type Deserializer struct {
+	blk  *Block
+	body *Body
```

Comment:
as a deserializer, it is more straightforward to have functions
`DeserializeBlock([]byte) (*Block, error)`, `ConvertFromBlockPb(*iotextypes.Block) (*Block, error)` and `DeserializeBlockBody([]byte) (body *BlockBody, error)`